rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -171,8 +171,64 @@ def _decode_vote_cast(self, context: DecoderContext) -> DecodingOutput:
 
 class ReloadableDecoderMixin(metaclass=ABCMeta):
 
+    def __init__(
+            self,
+            cache_type_to_check_for_freshness: CacheType,
+            query_data_method: Callable,
+            save_data_to_cache_method: Callable,
+            read_pools_and_gauges_from_cache_method: Callable,
+    ) -> None:
```

Comment:
I make a similar comment to what happened in the big velodrome PR. This interface is tough against porotocls with pools and gauges but `ReloadableDecoderMixin` as it was before can be used for anything for example yearn vaults or anything that might require to reload datal. I would create a subinterface that implements what you did

Reply:
I'm not sure about the value of data types for the other callables here. mypy seems to not handle them very well yet. it requires type:ignone wehn passing the methods as arguments. 
![image](https://github.com/rotki/rotki/assets/12560596/66588bb8-fc82-42ba-b7f8-7b2cedbf52d9)